test/utils/ui_state/user_value/volume_test.py

```diff
@@ -0,0 +1,172 @@
+    """
+    The function to test a use case of the Volume class:
+        User enters "3"
+        User enters "."
+        User enters "."
+        User enters "1"
+        User backspaces
+        User backspaces
+        User clears
+        User accepts
+    """
+    volume = Volume(Titrator(), UpdateSettings(Titrator(), MainMenu(Titrator())))
+
+    volume.loop()
+    lcd_out_mock.assert_has_calls(
+        [
+            mock.call(volume.get_label, line=1),
+            mock.call("", style=2, line=2),
+            mock.call("* = .       B = BS", line=3),
+            mock.call("A = accept  C = Clr", line=4),
+        ]
+    )
+
+    volume.handle_key("3")
+    assert volume.value == "3"
+
+    volume.loop()
+    lcd_out_mock.assert_has_calls(
+        [
+            mock.call(volume.get_label, line=1),
+            mock.call("3", style=2, line=2),
+            mock.call("* = .       B = BS", line=3),
+            mock.call("A = accept  C = Clr", line=4),
+        ]
+    )
+
+    volume.handle_key("*")
+    assert volume.value == "3."
+
+    volume.loop()
+    lcd_out_mock.assert_has_calls(
+        [
+            mock.call(volume.get_label, line=1),
+            mock.call("3.", style=2, line=2),
+            mock.call("* = .       B = BS", line=3),
+            mock.call("A = accept  C = Clr", line=4),
+        ]
+    )
+
+    volume.handle_key("*")
+    assert volume.value == "3."
+
+    volume.loop()
+    lcd_out_mock.assert_has_calls(
+        [
+            mock.call(volume.get_label, line=1),
+            mock.call("3.", style=2, line=2),
+            mock.call("* = .       B = BS", line=3),
+            mock.call("A = accept  C = Clr", line=4),
+        ]
+    )
+
+    volume.handle_key("1")
+    assert volume.value == "3.1"
+
+    volume.loop()
+    lcd_out_mock.assert_has_calls(
+        [
+            mock.call(volume.get_label, line=1),
+            mock.call("3.1", style=2, line=2),
+            mock.call("* = .       B = BS", line=3),
+            mock.call("A = accept  C = Clr", line=4),
+        ]
+    )
+
+    volume.handle_key("B")
+    assert volume.value == "3."
+
+    volume.loop()
+    lcd_out_mock.assert_has_calls(
+        [
+            mock.call(volume.get_label, line=1),
+            mock.call("3.", style=2, line=2),
+            mock.call("* = .       B = BS", line=3),
+            mock.call("A = accept  C = Clr", line=4),
+        ]
+    )
+
+    volume.handle_key("B")
+    assert volume.value == "3"
+
+    volume.loop()
+    lcd_out_mock.assert_has_calls(
+        [
+            mock.call(volume.get_label, line=1),
+            mock.call("3", style=2, line=2),
+            mock.call("* = .       B = BS", line=3),
+            mock.call("A = accept  C = Clr", line=4),
+        ]
+    )
+
+    volume.handle_key("C")
+    assert volume.value == ""
+
+    volume.loop()
+    lcd_out_mock.assert_has_calls(
+        [
+            mock.call(volume.get_label, line=1),
```

Comment:
I was expecting to see a string here. It maybe because you did not add the parentheses. The test should check the string output, not the which function is called.